nav2_loopback_sim/nav2_loopback_sim/utils.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2024 Open Navigation LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from geometry_msgs.msg import Quaternion, Transform
+import numpy as np
+import tf_transformations
+
+
+"""
+Transformation utilities for the loopback simulator
+"""
+
+
+def addYawToQuat(quaternion, yaw_to_add):
```

Comment:
Feel free to ignore below.  I know its 2D and simple.

You could make a quaternion from the yaw and then multiply it to the existing quaternion. 
In c++ I would also give it a normalize()   

https://docs.ros.org/en/jazzy/Tutorials/Intermediate/Tf2/Quaternion-Fundamentals.html#applying-a-quaternion-rotation

It seems that tf_transformations does not expose a quaternion normalisation.
https://github.com/DLu/tf_transformations/blob/07d77223288c457cfe71f361257e9fdd350c6fd9/tf_transformations/__init__.py#L809

Underneath it uses transforms3d.quaternions.qmult - http://matthew-brett.github.io/transforms3d/reference/transforms3d.quaternions.html#transforms3d.quaternions.qmult

That does provide a transforms3d.quaternions.qnorm - http://matthew-brett.github.io/transforms3d/reference/transforms3d.quaternions.html#qnorm

Reply:
Good to know! I didn't previously see the quaternion_multiply method in that library to use https://github.com/DLu/tf_transformations/blob/main/tf_transformations/__init__.py#L809

Will make that update